### ISSUE TITLE
test failure on 2024.01.1

### ISSUE THREAD

**Issue by amckinstry** [pydata/xarray#8692]
### What happened?

I'm the Debian maintainer of xarray. I'm packaging 2024.01.1, and. running tests we see 4 failures on python3.11.

### What did you expect to happen?

No failures in unit tests

### MVCE confirmation

- [X] Minimal example — the example is as focused as reasonably possible to demonstrate the underlying issue in xarray.
- [ ] Complete example — the example is self-contained, including all data and the text of any traceback.
- [ ] Verifiable example — the example copy & pastes into an IPython prompt or [Binder notebook](https://mybinder.org/v2/gh/pydata/xarray/main?urlpath=lab/tree/doc/examples/blank_template.ipynb), returning the result.
- [X] New issue — a search of GitHub Issues suggests this is not a duplicate.
- [X] Recent environment — the issue occurs with the latest version of xarray and its dependencies.

### Relevant log output

```Python
On python3.11, python3.12 we see:

FAILED xarray/tests/test_units.py::TestVariable::test_isel[int64-single value-single indexer-True] - TypeError: ones_like() got an unexpected keyword argument 'subok'
FAILED xarray/tests/test_units.py::TestVariable::test_isel[int64-multiple values-single indexer-True] - TypeError: ones_like() got an unexpected keyword argument 'subok'
FAILED xarray/tests/test_units.py::TestVariable::test_isel[int64-single value-multiple indexers-True] - TypeError: ones_like() got an unexpected keyword argument 'subok'
FAILED xarray/tests/test_units.py::TestVariable::test_isel[int64-multiple values-multiple indexers-True] - TypeError: ones_like() got an unexpected keyword argument 'subok'
```
This in detail looks like:
```

>       return da_func(*args, **kwargs)
E       TypeError: ones_like() got an unexpected keyword argument 'subok'

/usr/lib/python3/dist-packages/dask/array/core.py:1760: TypeError
```
so in. test_units.py its passing `subok` from xarray into `numpy.ones_like` for some reason.
```


### Anything else we need to know?

The code is at :

https://salsa.debian.org/science-team/python-xarray/-/tree/debian/latest?ref_type=heads


### Environment

Additional packages/plugins used:
<details>
Get: 1 http://deb.debian.org/debian sid/main arm64 libc-ares2 arm64 1.26.0-1 [140 kB]
Get: 2 http://deb.debian.org/debian sid/main arm64 python-hypothesis-doc all 6.93.2-1 [511 kB]
Get: 3 http://deb.debian.org/debian sid/main arm64 python-scipy-doc all 1.10.1-10 [33.4 MB]
Get: 4 http://deb.debian.org/debian sid/main arm64 python3-asciitree all 0.3.3-4 [5612 B]
Get: 5 http://deb.debian.org/debian sid/main arm64 python3-more-itertools all 10.2.0-1 [59.9 kB]
Get: 6 http://deb.debian.org/debian sid/main arm64 python3-scipy arm64 1.10.1-10 [17.8 MB]
Get: 7 http://deb.debian.org/debian sid/main arm64 python3-markupsafe arm64 2.1.4-1 [14.6 kB]
Get: 8 http://deb.debian.org/debian sid/main arm64 python3-xarray all 2023.12.0-3 [691 kB]
Get: 9 http://deb.debian.org/debian sid/main arm64 python3-monotonic all 1.6-2 [5728 B]
Get: 10 http://deb.debian.org/debian sid/main arm64 python3-fasteners all 0.18-2 [14.0 kB]
Get: 11 http://deb.debian.org/debian sid/main arm64 python3-sortedcontainers all 2.4.0-2 [31.9 kB]
Get: 12 http://deb.debian.org/debian sid/main arm64 python3-hypothesis all 6.93.2-1 [294 kB]
Get: 13 http://deb.debian.org/debian sid/main arm64 python3-numcodecs arm64 0.12.1+ds-1+b1 [424 kB]
Get: 14 http://deb.debian.org/debian sid/main arm64 python3-pydata-sphinx-theme all 0.7.2-3 [1075 kB]
Get: 15 http://deb.debian.org/debian sid/main arm64 python3-sphinx-copybutton all 0.4.0-4 [16.0 kB]
Get: 16 http://deb.debian.org/debian sid/main arm64 python3-sphinx-gallery all 0.10.1-4 [122 kB]
Get: 17 http://deb.debian.org/debian sid/main arm64 python3-sphinx-inline-tabs all 2023.4.21-1 [35.9 kB]
Get: 18 http://deb.debian.org/debian sid/main arm64 python3-zarr all 2.16.1+ds-2 [1311 kB]


</details>

**Comment by keewis** [comment]
this is known, and kind of unfortunate. `pint` used to explicitly pass `subok` to anything it wraps in `ones_like`, thus if something it wraps (like `dask`) does not accept that it will fail with the error you're seeing. This has been fixed in `pint>=0.21`, so you might want to require that?

**Comment by amckinstry** [comment]
Thanks.
